android/src/main/java/com/appcuesreactnative/AppcuesReactNativeModule.kt

```diff
@@ -23,57 +23,77 @@ class AppcuesReactNativeModule(reactContext: ReactApplicationContext) : ReactCon
     }
 
     @ReactMethod
-    fun setup(accountID: String, applicationID: String, options: ReadableMap?, additionalAutoProperties: ReadableMap?) {
+    fun setup(
+      accountID: String,
+      applicationID: String,
+      options: ReadableMap?, additionalAutoProperties: ReadableMap?,
+      promise: Promise
+    ) {
         val context = reactApplicationContextIfActiveOrWarn
+        if (context == null) {
+            promise.reject("no-context", "unable to initialize the SDK, no Application Context found")
+            return
+        }
```

Comment:
the only real change here is moving the null checks up top and rejecting the promise if needed, then resolving the promise at the end of the function in the normal success case.

other changes here are just whitespace/formatting as the conditional check and nesting was moved.